src/realm/sync/noinst/pending_bootstrap_store.cpp

```diff
@@ -0,0 +1,293 @@
+
+PendingBootstrapStore::PendingBootstrapStore(DBRef db, util::Logger* logger)
+    : m_db(std::move(db))
+    , m_logger(logger)
+{
+    std::vector<SyncMetadataTable> internal_tables{
+        {&m_table,
+         c_pending_bootstrap_table,
+         {&m_query_version, c_pending_bootstrap_query_version, type_Int},
+         {
+             {&m_changesets, c_pending_bootstrap_changesets, c_pending_changesets_table, true},
+             {&m_progress, c_pending_bootstrap_progress, c_progress_table, false},
+         }},
+        {&m_progress_table,
+         c_progress_table,
+         SyncMetadataTable::IsEmbeddedTag{},
+         {
+             {&m_progress_upload_server_version, c_progress_upload_server_version, type_Int},
+             {&m_progress_upload_client_version, c_progress_upload_client_version, type_Int},
+             {&m_progress_download_server_version, c_progress_download_server_version, type_Int},
+             {&m_progress_download_client_version, c_progress_download_client_version, type_Int},
+             {&m_progress_latest_server_version, c_progress_latest_server_version, type_Int},
+             {&m_progress_latest_server_version_salt, c_progress_latest_server_version_salt, type_Int},
+         }},
+        {&m_changeset_table,
+         c_pending_changesets_table,
+         SyncMetadataTable::IsEmbeddedTag{},
+         {
+             {&m_changeset_remote_version, c_pending_changesets_remote_version, type_Int},
+             {&m_changeset_last_integrated_client_version, c_pending_changesets_last_integrated_client_version,
+              type_Int},
+             {&m_changeset_origin_file_ident, c_pending_changesets_origin_file_ident, type_Int},
+             {&m_changeset_origin_timestamp, c_pending_changesets_origin_timestamp, type_Int},
+             {&m_changeset_original_changeset_size, c_pending_changesets_original_size, type_Int},
+             {&m_changeset_data, c_pending_changesets_data, type_Binary, true},
+         }}};
+
+    auto tr = m_db->start_read();
+    SyncMetadataSchemaVersions schema_versions(tr);
+    if (auto schema_version = schema_versions.get_version_for(tr, internal_schema_groups::c_pending_bootstraps)) {
+        if (*schema_version != c_schema_version) {
+            throw std::runtime_error("Invalid schema version for FLX sync pending bootstrap table group");
+        }
+        load_sync_metadata_schema(tr, &internal_tables);
+    }
+    else {
+        tr->promote_to_write();
+        create_sync_metadata_schema(tr, &internal_tables);
+        schema_versions.set_version_for(tr, internal_schema_groups::c_pending_bootstraps, c_schema_version);
+        tr->commit_and_continue_as_read();
+    }
+
+    // If there are any incomplete bootstraps in the table, we need to clear them and then update our has_pending
+    // flag accordingly.
+    if (auto bootstrap_table = tr->get_table(m_table); !bootstrap_table->is_empty()) {
+        m_has_pending = (bootstrap_table->is_empty() == false);
```

Comment:
doesn't the check above conclude that bootstrap_table is not empty?

Reply:
Fixed. also removed the comment above since that gets handled elsewhere now.